File: backend/app/translate/prompts/case_generation.py
"""Phase 2 固定窗口 Case 归纳 Prompt 入口"""

from __future__ import annotations

from .loader import load_prompt_md


def build_phase2_window_system_prompt(system_prompt_override: str | None = None) -> str:
    """构建 Phase 2 单窗口 System Prompt；override 来自 skill_hub resolve。"""
    if system_prompt_override:
        return system_prompt_override
    return load_prompt_md("steps-2-cases-skill.md")


def build_phase2_window_user_prompt(window_steps_plain_text: str, index_list_text: str) -> str:
    """构建 Phase 2 单窗口 User Prompt"""
    return f"""本窗口底层步骤记录（纯文本）：

{window_steps_plain_text}

本窗口可用 index 列表（consumeStepCount 对应前缀连续子集，从这里取）：{index_list_text}

请归纳成 1 个 Case，输出 Markdown 正文，最后一行输出 <case_meta consumeStepCount="N" lastIndex="..."/>。禁止 JSON。"""

File: feature_recorder/src/case_translate/prompts/case-generation.js
/**
 * case-generation.js - Phase 2 固定窗口 Case 归纳 Prompt 入口
 *
 * Skill：prompts/md/steps-2-cases-skill.md
 * User：本文件内嵌（动态数据拼接）
 */

import { loadPromptMd } from './loader.js';

/**
 * 构建 Phase 2 单窗口 System Prompt
 *
 * @returns {string}
 */
export function buildPhase2WindowSystemPrompt() {
  return loadPromptMd('steps-2-cases-skill.md');
}

/**
 * 构建 Phase 2 单窗口 User Prompt（纯文本步骤流）
 *
 * @param {string} windowStepsPlainText - 当前窗口步骤纯文本
 * @param {string} indexListText - 本窗口 index 列表 JSON 字符串
 * @returns {string}
 */
export function buildPhase2WindowUserPrompt(windowStepsPlainText, indexListText) {
  return `本窗口底层步骤记录（纯文本）：

${windowStepsPlainText}

本窗口可用 index 列表（consumeStepCount 对应前缀连续子集，从这里取）：${indexListText}

请归纳成 1 个 Case，输出 Markdown 正文，最后一行输出 <case_meta consumeStepCount="N" lastIndex="..."/>。禁止 JSON。`;
}
